components/calendar/src/indian.rs

```diff
@@ -0,0 +1,179 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! This module contains types and implementations for the Indian national calendar
+
+use crate::iso::{Iso, IsoYear};
+use crate::{
+    types, ArithmeticDate, Calendar, CalendarArithmetic, Date, DateDuration, DateDurationUnit,
+    DateTime, DateTimeError,
+};
+use core::marker::PhantomData;
+use tinystr::tinystr;
+
+#[derive(Copy, Clone, Debug, Hash, Default, Eq, PartialEq)]
+// The Indian national calendar
+pub struct Indian;
+
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq)]
+pub struct IndianDateInner(ArithmeticDate<Indian>);
+
+impl CalendarArithmetic for Indian {
+    fn month_days(year: i32, month: u8) -> u8 {
+        if month == 1 {
+            if Self::is_leap_year(year) {
+                31
+            } else {
+                30
+            }
+        } else if (2..=6).contains(&month) {
+            31
+        } else if (7..=12).contains(&month) {
+            30
+        } else {
+            0
+        }
+    }
+
+    fn months_for_every_year() -> u8 {
+        12
+    }
+
+    fn is_leap_year(year: i32) -> bool {
+        Iso::is_leap_year(IsoYear(year + 78))
+    }
+}
+
+impl Calendar for Indian {
+    type DateInner = IndianDateInner;
+    fn date_from_iso(&self, iso: Date<Iso>) -> IndianDateInner {
+        let day_of_year = Iso::day_of_year(*iso.inner());
+        IndianDateInner(Self::date_from_year_day(
```

Comment:
@Manishearth I want to make this part of calendar_arithmetic as well, seems like a good function to be generalized. But I think I kinda did it wrong?! Lemme know whats the right way to add this to calendar_arithmetic

Reply:
this = `date_from_year_day(year: i32, year_day: u32) -> Date<C>`

Reply:
One way of doing this would be, for the method to create an ArithmeticCalendar variable, and call from that. If thats the only way, can do that. If there is a better way, do lemme know

Reply:
I think you can add a `fn date_from_year_day(&self, year, day) -> ArithmeticDate<C>`? You'd want to create an ArithmeticCalendar.

Reply:
Seems kinda wrong that, to call a method that creates an ArithmeticDate, we need to create an ArithemeticDate

Reply:
Oh sorry, the method would be `ArithmeticDate::new_from_year_day(&calendar, year, day)`